On uninstallation of module, delete any keys that use Lockr as the key provider

rlhawk opened the pull request:
I added an implementation of hook_uninstall() that deletes keys that use Lockr as the key provider, as well as the key values within Lockr.

rlhawk committed:
On uninstallation of module, delete any keys that use Lockr as the key provider.

tynor commented on pantheon/drupal7/lockr_pantheon/lockr_pantheon.install:
```diff
@@ -0,0 +1,46 @@
+ */
+function lockr_pantheon_uninstall() {
+  // Load the key configurations.
+  $configs = db_query("SELECT * FROM {key_config} ORDER BY label ASC")->fetchAllAssoc('id', PDO::FETCH_ASSOC);
```
For the future, we might want to wrap this query (as well as the loop below) into a function for key like `key_get_keys_for_provider($provider_name)` so it's easier for providers to implement this kind of feature.

rlhawk commented:
I think it's better just to do it with a database call, so Key doesn't need to be involved. The fewer modules we have to manually load with drupal_load(), the better.

tynor commented:
Actually, I don't think I can modify the PR content, but the for loop filtering out Lockr keys can be moved into the query itself so we aren't needlessly pulling all rows out of the database:

``` php
$configs = db_query("SELECT * FROM {key_config} WHERE key_provider = :provider ORDER BY label ASC", array('provider' => 'config'))->fetchAllAssoc('id', PDO::FETCH_ASSOC);
```

Probably doesn't really matter since people won't likely have thousands of keys or more, but still probably good to not do unnecessary work.

rlhawk commented:
Oh, of course. I'll make that change.

tynor commented:
Okay, yeah, not loading key is good—forgot that key might not exist.

rlhawk committed:
Simplified retrieval of Lockr keys from database.

rlhawk commented:
I updated the code. In addition to adding the WHERE clause to the query, I removed the ORDER BY, since we don't need the returned records to be sorted.

I also updated the Lockr Demo site.

rlhawk committed:
Use the correct provider when retrieving Lockr keys.
